pkg/controllers/machine/disruption/drift.go

```diff
@@ -60,7 +62,7 @@ func (d *Drift) Reconcile(ctx context.Context, provisioner *v1alpha5.Provisioner
 		return reconcile.Result{}, cloudprovider.IgnoreMachineNotFoundError(fmt.Errorf("getting drift for machine, %w", err))
 	}
 	// 3. Otherwise, if the machine isn't drifted, but has the status condition, remove it.
-	if !drifted {
+	if drifted == "" {
```

Comment:
Isn't the point of using Enums so that we can check if it's `drifted == NotDrifted`?